File: Difficulty: Easy/Isomorphic Strings/isomorphic-strings.py
#User function Template for python3

class Solution:
    
    #Function to check if two strings are isomorphic.
    def areIsomorphic(self,s1,s2):
        l1=[]
        l2=[]
        for i in s1:
            l1.append(s1.index(i))
        for i in s2:
            l2.append(s2.index(i))
        if l1==l2:
            return True
        return False


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys
from collections import defaultdict

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER


@atexit.register
def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())


if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        s = str(input())
        p = str(input())
        ob = Solution()
        if (ob.areIsomorphic(s, p)):
            print("true")
        else:
            print("false")
        print("~")

# } Driver Code Ends

File: Difficulty: Easy/Isomorphic Strings/isomorphic-strings.java
//{ Driver Code Starts
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Arrays;

public class Driver {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine());
        while (tc-- > 0) {
            String s1 = br.readLine();
            String s2 = br.readLine();

            Solution obj = new Solution();

            boolean a = obj.areIsomorphic(s1, s2);
            if (a)
                System.out.println("true");
            else
                System.out.println("false");

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


class Solution {
    // Function to check if two strings are isomorphic.
    public static boolean areIsomorphic(String s1, String s2) {
        // Your code here

        List<Integer> l1 = new ArrayList<>();
        List<Integer> l2 = new ArrayList<>();
        
        // Loop through the first string and store the index of the first occurrence of each character.
        for (char c : s1.toCharArray()) {
            l1.add(s1.indexOf(c));
        }
        
        // Loop through the second string and store the index of the first occurrence of each character.
        for (char c : s2.toCharArray()) {
            l2.add(s2.indexOf(c));
        }
        
        // Compare the two lists.
        if (l1.equals(l2)) {
            return true;
        }
        return false;
    }

   
}